tock/api/views.py

```diff
@@ -4,44 +4,94 @@
 from django.db.models import Count, Sum
 from decimal import Decimal
 
-from django.contrib.auth.models import User, Group
-from hours.models import ReportingPeriod, Timecard, TimecardObject
-from projects.models import Project, Agency, AccountingCode
+from django.contrib.auth.models import User
+from projects.models import Project
+from hours.models import TimecardObject
 
 from rest_framework import serializers, generics, pagination, renderers
 
-from rest_framework.renderers import JSONRenderer
-from .renderers import PaginatedCSVRenderer
-
 import csv
+from .renderers import PaginatedCSVRenderer, stream_csv
 
 class StandardResultsSetPagination(pagination.PageNumberPagination):
+    """
+    This is a standard results set paginator for all API view classes
+    that need pagination.
+    """
     page_size = 100
     page_size_query_param = 'page_size'
     max_page_size = 500
 
+
+# Serializers for different models
+
 class ProjectSerializer(serializers.ModelSerializer):
-    billable = serializers.BooleanField(source='accounting_code.billable')
     class Meta:
         model = Project
-        fields = ('id', 'name', 'description', 'billable',)
+        fields = (
+            'id',
+            'name',
+            'description',
+            'billable',
+        )
+    billable = serializers.BooleanField(source='accounting_code.billable')
 
 class UserSerializer(serializers.ModelSerializer):
     class Meta:
         model = User
-        fields = ('id', 'username', 'first_name', 'last_name',)
+        fields = (
+            'id',
+            'username',
+            'first_name',
+            'last_name',
+        )
 
 class TimecardSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = TimecardObject
+        fields = (
+            'user',
+            'project_name',
+            'project_id',
+            'start_date',
+            'end_date',
+            'hours_spent',
+            'billable',
+        )
     user = serializers.StringRelatedField(source='timecard.user')
     project_id = serializers.CharField(source='project.id')
     project_name = serializers.CharField(source='project.name')
     start_date = serializers.DateField(source='timecard.reporting_period.start_date')
     end_date = serializers.DateField(source='timecard.reporting_period.end_date')
     billable = serializers.BooleanField(source='project.accounting_code.billable')
+
+class BulkTimecardSerializer(serializers.ModelSerializer):
```

Comment:
This is very similar to `TimecardSerializer`. Could these classes be merged, or share a base class?